src/libraries/System.Formats.Tar/tests/TarReader/TarReader.File.Tests.Base.cs

```diff
@@ -0,0 +1,620 @@
+                VerifyGlobalExtendedAttributes(reader);
+            }
+
+            TarEntry file = reader.GetNextEntry();
+
+            VerifyRegularFileEntry(file, format, "file.txt", $"Hello {testCaseName}");
+
+            TarEntry hardLink = reader.GetNextEntry();
+            // The 'tar' tool detects hardlinks as regular files and saves them as such in the archives, for all formats
+            VerifyRegularFileEntry(hardLink, format, "hardlink.txt", $"Hello {testCaseName}");
+
+            Assert.Null(reader.GetNextEntry());
+        }
+
+        protected void Read_Archive_File_SymbolicLink_Internal(TarEntryFormat format, TestTarFormat testFormat)
+        {
+            string testCaseName = "file_symlink";
+            using MemoryStream ms = GetTarMemoryStream(CompressionMethod.Uncompressed, testFormat, testCaseName);
+
+            using TarReader reader = new TarReader(ms);
+
+            if (testFormat is TestTarFormat.pax_gea)
+            {
+                VerifyGlobalExtendedAttributes(reader);
+            }
+
+            TarEntry file = reader.GetNextEntry();
+
+            VerifyRegularFileEntry(file, format, "file.txt", $"Hello {testCaseName}");
+
+            TarEntry symbolicLink = reader.GetNextEntry();
+            VerifySymbolicLinkEntry(symbolicLink, format, "link.txt", "file.txt");
+
+            Assert.Null(reader.GetNextEntry());
+        }
+
+        protected void Read_Archive_Folder_File_Internal(TarEntryFormat format, TestTarFormat testFormat)
+        {
+            string testCaseName = "folder_file";
+            using MemoryStream ms = GetTarMemoryStream(CompressionMethod.Uncompressed, testFormat, testCaseName);
+
+            using TarReader reader = new TarReader(ms);
+
+            if (testFormat is TestTarFormat.pax_gea)
+            {
+                VerifyGlobalExtendedAttributes(reader);
+            }
+
+            TarEntry directory = reader.GetNextEntry();
+
+            VerifyDirectoryEntry(directory, format, "folder/");
+
+            TarEntry file = reader.GetNextEntry();
+            VerifyRegularFileEntry(file, format, "folder/file.txt", $"Hello {testCaseName}");
+
+            Assert.Null(reader.GetNextEntry());
+        }
+
+        protected void Read_Archive_Folder_File_Utf8_Internal(TarEntryFormat format, TestTarFormat testFormat)
+        {
+            string testCaseName = "folder_file_utf8";
+            using MemoryStream ms = GetTarMemoryStream(CompressionMethod.Uncompressed, testFormat, testCaseName);
+
+            using TarReader reader = new TarReader(ms);
+
+            if (testFormat is TestTarFormat.pax_gea)
+            {
+                VerifyGlobalExtendedAttributes(reader);
+            }
+
+            TarEntry directory = reader.GetNextEntry();
+
+            VerifyDirectoryEntry(directory, format, "földër/");
+
+            TarEntry file = reader.GetNextEntry();
+            VerifyRegularFileEntry(file, format, "földër/áöñ.txt", $"Hello {testCaseName}");
```

Comment:
For best results, don't check in unicode characters, use the \uxxxx representation instead.

Reply:
By the by, this is style rule 16:

> When including non-ASCII characters in the source code use Unicode escape sequences (\uXXXX) instead of literal characters. Literal non-ASCII characters occasionally get garbled by a tool or editor.

https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/coding-style.md

Reply:
Fixed